agent/src/main/resources/build-scan-init.gradle

```diff
@@ -21,21 +53,49 @@ def buildScanPublishedAction = { def buildScan ->
 if (GradleVersion.current() < GradleVersion.version('6.0')) {
     // Gradle 4.1+ and 5.+
     rootProject {
+        if (autoApplyGe && !pluginManager.hasPlugin(BUILD_SCAN_PLUGIN_ID)) {
+            pluginManager.apply(initscript.classLoader.loadClass(BUILD_SCAN_PLUGIN_CLASS))
+        }
+
+        if (autoApplyCcud && !pluginManager.hasPlugin(CCUD_PLUGIN_ID)) {
+            pluginManager.apply(initscript.classLoader.loadClass(CCUD_PLUGIN_CLASS))
+        }
+
         pluginManager.withPlugin(BUILD_SCAN_PLUGIN_ID) {
             buildScanPublishedAction(buildScan)
+            buildScan {
+                afterEvaluate {
+                    server = geUrl ?: server
+                    publishAlwaysIf(autoApplyGe)
+                }
+            }
         }
     }
 } else {
     // Gradle 6.+
     gradle.settingsEvaluated { settings ->
-        extensionsWithPublicType(settings, GRADLE_ENTERPRISE_EXTENSION_CLASS).each {
-            buildScanPublishedAction(settings[it.name].buildScan)
+        if (autoApplyGe && !settings.pluginManager.hasPlugin(GRADLE_ENTERPRISE_PLUGIN_ID)) {
+            settings.pluginManager.apply(initscript.classLoader.loadClass(GRADLE_ENTERPRISE_PLUGIN_CLASS))
+        }
+
+        if (autoApplyCcud && !settings.pluginManager.hasPlugin(CCUD_PLUGIN_ID)) {
+            settings.pluginManager.apply(initscript.classLoader.loadClass(CCUD_PLUGIN_CLASS))
+        }
+
+        extensionsWithPublicType(settings, GRADLE_ENTERPRISE_EXTENSION_CLASS).each { ext ->
+            buildScanPublishedAction(ext.buildScan)
+            ext.server = geUrl ?: ext.server
+            ext.buildScan {
+                publishAlwaysIf(autoApplyGe)
+            }
         }
     }
 }
 
 static def extensionsWithPublicType(def container, String publicType) {
-    container.extensions.extensionsSchema.elements.findAll { it.publicType.concreteClass.name == publicType }
+    container.extensions.extensionsSchema.elements
+            .findAll { it.publicType.concreteClass.name == publicType }
+            .collect { container[it.name] }
 }
```

Comment:
Can you explain why this change was needed?

Reply:
It wasn't, and I reverted on this update. I think this stuck around through some earlier iterations.